Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -1,30 +1,100 @@
+using System.Linq;
+using System.Collections.Generic;
 
 namespace Tavisca.Bootcamp.LanguageBasics.Exercise1
 {
-    public static class Program
-    {
-        static void Main(string[] args)
-        {
-            Test(new[] {"12:12:12"}, new [] { "few seconds ago" }, "12:12:12");
-            Test(new[] { "23:23:23", "23:23:23" }, new[] { "59 minutes ago", "59 minutes ago" }, "00:22:23");
-            Test(new[] { "00:10:10", "00:10:10" }, new[] { "59 minutes ago", "1 hours ago" }, "impossible");
-            Test(new[] { "11:59:13", "11:13:23", "12:25:15" }, new[] { "few seconds ago", "46 minutes ago", "23 hours ago" }, "11:59:23");
-            Console.ReadKey(true);
-        }
-
-        private static void Test(string[] postTimes, string[] showTimes, string expected)
-        {
-            var result = GetCurrentTime(postTimes, showTimes).Equals(expected) ? "PASS" : "FAIL";
-            var postTimesCsv = string.Join(", ", postTimes);
-            var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
-        }
-
-        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
-        }
-    }
+	public static class Program
+	{
+		static void Main(string[] args)
+		{
+			Test(new[] {"12:12:12"}, new [] { "few seconds ago" }, "12:12:12");
+			Test(new[] { "23:23:23", "23:23:23" }, new[] { "59 minutes ago", "59 minutes ago" }, "00:22:23");
+			Test(new[] { "00:10:10", "00:10:10" }, new[] { "59 minutes ago", "1 hours ago" }, "impossible");
+			Test(new[] { "11:59:13", "11:13:23", "12:25:15" }, new[] { "few seconds ago", "46 minutes ago", "23 hours ago" }, "11:59:23");
+           // Console.ReadKey(true);
+		}
+
+		private static void Test(string[] postTimes, string[] showTimes, string expected)
+		{
+			var result = GetCurrentTime(postTimes, showTimes).Equals(expected) ? "PASS" : "FAIL";
+			var postTimesCsv = string.Join(", ", postTimes);
+			var showTimesCsv = string.Join(", ", showTimes);
+			Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+		}
+
+		public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+		{
+
+         /************************************************************
+          Idea is to solve this problem is to find innermost subset of the 
+			time and return the first valid  time in that subset 
+			 # 86400 means 24 hour
+			 # 3599 means 3600s - 1s
+			 # 59 means 60 s -1s
+
+
+		**********************************************************************/
+         // early detection of no solution
+			for(int i=0;i< exactPostTime.Length;i++){
+
+				for(int j=i+1;j<exactPostTime.Length;j++){
+					if(exactPostTime[i].Equals(exactPostTime[j]))
+					if(!showPostTime[i].Equals(showPostTime[j]))
+					return "impossible";
+
+				}
+			}
+
+			int totalsec,low,high;

```

Comment:
instead of taking sol_array_1 this as integer, can we take them as TimeSpan so that conversion of int to time span is fairly easy and straight forward.